tasks/main.yml

```diff
@@ -1,39 +1,50 @@
 ---
 - include_tasks: debian.yml
   when: ansible_os_family == 'Debian'
+- name: make port 80 available for certbot standalone to work
+  block:
+    - name: Stop webserver
+      service:
+        name: "{{ certbot_webserver_installed }}"
+        state: stopped
+      ignore_errors: yes
 
-- name: Stop webserver
-  service:
-    name: "{{ certbot_webserver_installed }}"
-    state: stopped
-  ignore_errors: yes
-
-- name: Check whether port 80 is available
-  wait_for:
-    port: 80
-    state: stopped
-    timeout: "{{ certbot_waitfor_port_seconds }}"
+    - name: Check whether port 80 is available
+      wait_for:
+        port: 80
+        state: stopped
+        timeout: "{{ certbot_waitfor_port_seconds }}"
+  when: certbot_plugin == "standalone"
 
 - name: Create certs
-  command: letsencrypt certonly --standalone -d {{ site_name }} -m {{ certbot_mail_address }} --agree-tos --noninteractive --text
+  command: certbot certonly --{{ certbot_plugin }} -d {{ site_name }} -m {{ certbot_mail_address }} --agree-tos --noninteractive --text
   args:
     creates: "{{ certbot_crt }}"
-  when: certbot_create_certs
+  when:
+    - certbot_create_certs
+    - not certbot_install_cert
+
+- name: Create and install cert using {{ certbot_plugin }} plugin
+  command: "certbot --{{ certbot_plugin }} {{ (certbot_install_cert) | ternary('','certonly') }} -d {{ site_name }} -m {{ certbot_mail_address }} --agree-tos --noninteractive --text"
+  when: certbot_install_cert
 
 - include_tasks: copy-files.yml
 
-- name: Renew certs every month
+- name: Create cron job to renew certs every month
   cron:
     name: renewCertbotCerts
     minute: "{{ certbot_renew_certs_minute }}"
     hour: "{{ certbot_renew_certs_hour }}"
     month: "{{ certbot_renew_certs_month }}"
     weekday: "{{ certbot_renew_certs_weekday }}"
-    job: /usr/bin/letsencrypt renew
-  when: certbot_create_certs
+    job: /usr/bin/certbot renew --prehook "service {{ certbot_webserver_installed }} stop" --post-hook "service {{ certbot_webserver_installed }} start"
+  when:
+    - certbot_create_certs
+    - certbot_plugin == "standalone"
```

Comment:
@morrismukiri does this mean that autorenewal will only happen if you use certbot in standalone mode?

Reply:
After autorenewal, will it be up to the NGINX role, for instance, to copy over the new files?

Reply:
certbot automatically creates a systemd timer that runs twice daily. The renew command uses  the same strategy used to generate the certificate. The reason standalone is special is because certbot won't know to stop the web server(as we do when creating the cert), so we create a cron job that is going to stop the web server and start it once renewal is done.

Reply:
The role assumes that if you don't want certbot to install the cert into the site configurations, you already have that figured out; either by copying the certs from the letsencrypt directory or using the path directly in the site configuration. Otherwise certbot will modify the site configuration to add the certs and redirect.

Reply:
Cool. Makes sense.